scarb/src/core/manifest/toml_manifest.rs

```diff
@@ -1275,19 +1306,48 @@ impl TomlManifest {
         Ok(profile)
     }
 
+    fn verify_toml_cairo_section(profile_definition: &TomlProfile, ui: Ui) -> Result<()> {
+        let Some(cairo) = &profile_definition.cairo else {
+            return Ok(());
+        };
+
+        if cairo.skip_optimizations == Some(true)
+            && cairo
+                .inlining_strategy
+                .as_ref()
+                .is_none_or(|inlining_strategy| inlining_strategy == &InliningStrategy::Avoid)
```

Comment:
```suggestion
            && !cairo
                .inlining_strategy
                .as_ref()
                .is_none_or(|inlining_strategy| inlining_strategy == &InliningStrategy::Avoid)
```
Isn't `None | "avoid"` actually what we want the user to use?